source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -203,6 +213,56 @@ to the ``FindOne()`` method, which performs the following actions:
 
       {"Item":"Masala","Rating":9,"DateOrdered":"2009-11-12T05:00:00Z"}
 
+.. _golang-find-one-by-id:
+
+Find One by Object ID Example
+`````````````````````````````
+
+The following example defines an ``id`` variable with an ``ObjectId``
+value and passes a context, filter, and ``FindOneOptions`` to the 
+``FindOne()`` method. Then, ``FindOne()`` performs the following actions:
+
+- Matches the document with the specified ``ObjectId`` value
+- Projects the ``Item`` and ``Rating`` fields of the matched document
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: go
+
+      id, err := primitive.ObjectIDFromHex("65170b42b99efdd0b07d42de")
+      if err != nil {
+          panic(err)
+      }
+
+      filter := bson.D{{"_id", id}}
+      opts := options.FindOne().SetProjection(
+          bson.D{{"item",1}, {"rating",1}, {"_id", 0}}
+      )
+
+      var result bson.D
+      err = coll.FindOne(context.TODO(), filter).Decode(&result)
+      if err != nil {
+          panic(err)
+      }
+
+      res, _ := json.Marshal(result)
+      fmt.Println(string(res))
+
+   .. output::
+      :language: none
+      :visible: false
+
+      [{"Key":"item","Value":"Hibiscus"},{"Key":"rating","Value":4}]
+
+.. note::
+
+   The {+driver-short+} automatically generates a unique ``ObjectId`` value for
+   each document's ``_id`` field. The preceding code example uses an ``ObjectID``
+   with a hexidecimal string representation of ``"65170b42b99efdd0b07d42de"``,
+   but your ``ObjectID`` values are unique.
```

Comment:
S: this note would be more helpful if it linked to some information about the `_id` field, like [this](https://docs-mongodbcom-staging.corp.mongodb.com/golang/docsworker-xlarge/DOCSP-32693-find-by-id/fundamentals/crud/write-operations/insert/#the-_id-field). I think the message of this sentence is just that users will have to know the Id of the doc they are looking for to search with this function -- and that message is kind of confusing right now